contracts/strategies/RFPStrategy.sol

```diff
@@ -0,0 +1,278 @@
+        }
+
+        Recipient memory recipient = Recipient({
+            recipientAddress: recipientAddress,
+            proposalBid: proposalBid,
+            recipientStatus: RecipientStatus.Pending
+        });
+
+        recipients[recipientId] = recipient;
+        _recipientIds.push(recipientId);
+
+        emit Registered(recipientId, _data, _sender);
+    }
+
+    /// @notice Select recipient for RFP allocation
+    /// @param _data The data to be decoded
+    /// @param _sender The sender of the allocation
+    function _allocate(bytes memory _data, address _sender) internal override nonReentrant onlyPoolManager(_sender) {
+        if (acceptedRecipientId != address(0)) {
+            revert RECIPIENT_ALREADY_ACCEPTED();
+        }
+
+        acceptedRecipientId = abi.decode(_data, (address));
+
+        uint256 _recipientsCounter = _recipientIds.length;
+        for (uint256 i = 0; i < _recipientsCounter;) {
+            // update status of all other recipients to rejected
+            address recipientId = _recipientIds[i];
+            if (recipientId != acceptedRecipientId) {
+                recipients[recipientId].recipientStatus = RecipientStatus.Rejected;
+            }
+
+            unchecked {
+                i++;
+            }
+        }
+
+        // update status of acceptedRecipientId to accepted
+        if (acceptedRecipientId != address(0)) {
+            Recipient storage recipient = recipients[acceptedRecipientId];
+            recipient.recipientStatus = RecipientStatus.Accepted;
+
+            IAllo.Pool memory pool = allo.getPool(poolId);
+
+            emit Allocated(acceptedRecipientId, recipient.proposalBid, pool.token, _sender);
+        }
+    }
+
+    function _distribute(address[] memory, bytes memory, address _sender) internal override onlyPoolManager(_sender) {
+        if (acceptedRecipientId == address(0)) {
+            revert NO_ACCEPTED_RECIPIENT();
+        }
+
+        if (upcomingMilestone > milestones.length) {
+            revert INVALID_MILESTONE();
+        }
+
+        IAllo.Pool memory pool = allo.getPool(poolId);
+        Milestone storage milestone = milestones[upcomingMilestone];
+        Recipient memory recipient = recipients[acceptedRecipientId];
+
+        uint256 amount = recipient.proposalBid * milestone.amountPercentage / 1e18;
+
+        allo.decreasePoolTotalFunding(poolId, amount);
+        _transferAmount(pool.token, recipient.recipientAddress, amount);
+
+        milestone.milestoneStatus = RecipientStatus.Accepted;
+        upcomingMilestone++;
+
+        emit Distributed(acceptedRecipientId, recipient.recipientAddress, amount, _sender);
+    }
+
+    function _isIdentityManager(address _anchor, address _sender) internal view returns (bool) {
+        IRegistry registry = allo.getRegistry();
+        IRegistry.Identity memory identity = registry.getIdentityByAnchor(_anchor);
+        return registry.isOwnerOrMemberOfIdentity(identity.id, _sender);
+    }
+}
```

Comment:
Missing functionalities:
When does the pool end?

WIthdrawing funds

Reply:
Do we need a pool ended variable  ? @nfrgosselin

Reply:
as a pool owner you might want to close the pool if you can't find a recipient which fits your needs.

also we should think about introducing functions on the interface for:

canRegister(address)
canAllocate(address)
.. or sth similar. It differs from isValidAllocator, since canAllocate could check if:
address == isValidAllocator && acceptedRecipientId == address (0) && ...
=> basically a helper function for frontend implementations

we talked earlier in one of our discussions about it.

Reply:
Confirmed from standup: we want to have a variable in the core Allo interface that indicates whether or not a pool is active